### PR TITLE
fix: Clear compaction tasks when release channel

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #31648
pr: #31666

### CODE REVIEW

**Comment by yanliang567** [comment]
rerun failed ut tests....

File: internal/datanode/compaction_executor.go [hunk 1]
```diff
@@ -120,22 +126,71 @@ func (c *compactionExecutor) channelValidateForCompaction(vChannelName string) b
 	return !c.dropped.Contain(vChannelName)
 }
 
-func (c *compactionExecutor) stopExecutingtaskByVChannelName(vChannelName string) {
-	c.dropped.Insert(vChannelName)
+func (c *compactionExecutor) discardByDroppedChannel(channel string) {
+	c.dropped.Insert(channel)
```
Comment (congqixia) [inline internal/datanode/compaction_executor.go:130]: it looks like the `dropped` vchannel is never removed
what if a channel is balanced to another datanode and back? this datanode could never be able to execute compaction anymore

Comment (XuanYang-cn) [reply internal/datanode/compaction_executor.go:130]: dropped will be mark as drop if and only if the collection is dropped.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**